cupy/core/_routines_statistics.pyx

```diff
@@ -55,18 +55,31 @@ cdef ndarray _ndarray_argmin(ndarray self, axis, out, dtype, keepdims):
 
 
 cdef ndarray _ndarray_mean(ndarray self, axis, dtype, out, keepdims):
+    dtype_sum = dtype_out = dtype
+    if dtype is None:
+        if self.dtype.kind in 'iub':
+            dtype_out = dtype_sum = numpy.float64
```

Comment:
Could you separate the assignments to two lines?

Reply:
okay